src/main/java/com/microsoft/jenkins/kubernetes/V1ResourceManager.java

```diff
@@ -0,0 +1,599 @@
+import io.kubernetes.client.models.V1ReplicationController;
+import io.kubernetes.client.models.V1Secret;
+import io.kubernetes.client.models.V1Service;
+import io.kubernetes.client.models.V1ServicePort;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class V1ResourceManager extends ResourceManager {
+    private CoreV1Api coreV1ApiInstance;
+    private AppsV1Api appsV1ApiInstance;
+    private BatchV1Api batchV1ApiInstance;
+    private String pretty = DEFAULT_PRETTY;
+    private V1ResourceUpdateMonitor resourceUpdateMonitor = V1ResourceUpdateMonitor.NOOP;
+
+    public V1ResourceManager() {
+        coreV1ApiInstance = new CoreV1Api();
+        appsV1ApiInstance = new AppsV1Api();
+        batchV1ApiInstance = new BatchV1Api();
+    }
+
+    public V1ResourceManager(String pretty) {
+        this();
+        this.pretty = pretty;
+    }
+
+    @Override
+    public boolean apply(Object resource) throws IOException {
+        if (resource instanceof V1Namespace) {
+            V1Namespace namespace = (V1Namespace) resource;
+            new NamespaceUpdater(namespace).createOrApply();
+        } else if (resource instanceof V1Deployment) {
+            V1Deployment deployment = (V1Deployment) resource;
+            new DeploymentUpdater(deployment).createOrApply();
+        } else if (resource instanceof V1Service) {
+            V1Service service = (V1Service) resource;
+            new ServiceUpdater(service).createOrApply();
+        } else if (resource instanceof V1ReplicationController) {
+            V1ReplicationController replicationController = (V1ReplicationController) resource;
+            new ReplicationControllerUpdater(replicationController).createOrApply();
+        } else if (resource instanceof V1ReplicaSet) {
+            V1ReplicaSet replicaSet = (V1ReplicaSet) resource;
+            new ReplicaSetUpdater(replicaSet).createOrApply();
+        } else if (resource instanceof V1DaemonSet) {
+            V1DaemonSet daemonSet = (V1DaemonSet) resource;
+            new DaemonSetUpdater(daemonSet).createOrApply();
+        } else if (resource instanceof V1Job) {
+            V1Job job = (V1Job) resource;
+            new JobUpdater(job).createOrApply();
+        } else if (resource instanceof V1Pod) {
+            V1Pod pod = (V1Pod) resource;
+            new PodUpdater(pod).createOrApply();
+        } else if (resource instanceof V1Secret) {
+            V1Secret secret = (V1Secret) resource;
+            new SecretUpdater(secret).createOrApply();
+        } else if (resource instanceof V1ConfigMap) {
+            V1ConfigMap configMap = (V1ConfigMap) resource;
+            new ConfigMapUpdater(configMap).createOrApply();
+        } else {
+            return false;
+        }
+        return true;
+    }
+
+    public V1ResourceUpdateMonitor getResourceUpdateMonitor() {
```

Comment:
Does these getResourceUpdateMonitor/withResourceUpdateMonitor ever used? I didn't find any invoker

Reply:
I think the former design is being able to use this methods to create a resourceManager with different monitor since the Monitor variable here is private. No invoker in our code because we use the default monitor here.